client/my-sites/site-settings/form-general.jsx

```diff
@@ -451,15 +444,37 @@ export class SiteSettingsFormGeneral extends Component {
 					) }
 				{ isWpcomStagingSite && (
 					<>
-						<PublicFormRadio />
+						{ this.publicFormRadio(
+							( wpcomPublicComingSoon && blogPublic === 0 && isComingSoonDisabled ) ||
+								( blogPublic === 0 && ! wpcomPublicComingSoon ) ||
+								blogPublic === 1,
+							() =>
+								this.handleVisibilityOptionChange( {
+									blog_public: isWpcomStagingSite ? 0 : 1,
+									wpcom_coming_soon: 0,
+									wpcom_public_coming_soon: 0,
+								} )
+						) }
 						<FormSettingExplanation>
 							{ translate(
 								'Your site is visible to everyone, but search engines are discouraged from indexing staging sites.'
 							) }
 						</FormSettingExplanation>
 					</>
 				) }
-				{ ! isNonAtomicJetpackSite && ! isWpcomStagingSite && <PublicFormRadio /> }
+				{ ! isNonAtomicJetpackSite &&
+					! isWpcomStagingSite &&
+					this.publicFormRadio(
+						( wpcomPublicComingSoon && blogPublic === 0 && isComingSoonDisabled ) ||
+							( blogPublic === 0 && ! wpcomPublicComingSoon ) ||
+							blogPublic === 1,
+						() =>
+							this.handleVisibilityOptionChange( {
+								blog_public: isWpcomStagingSite ? 0 : 1,
+								wpcom_coming_soon: 0,
+								wpcom_public_coming_soon: 0,
+							} )
+					) }
```

Comment:
After analyzing the code, my suggestion would be to remove one of the conditions:

```
				{ ! isNonAtomicJetpackSite && (
					<>
						<FormLabel className="site-settings__visibility-label is-public">
							<FormRadio
								name="blog_public"
								value="1"
								checked={
									( wpcomPublicComingSoon && blogPublic === 0 && isComingSoonDisabled ) ||
									( blogPublic === 0 && ! wpcomPublicComingSoon ) ||
									blogPublic === 1
								}
								onChange={ () =>
									this.handleVisibilityOptionChange( {
										blog_public: isWpcomStagingSite ? 0 : 1,
										wpcom_coming_soon: 0,
										wpcom_public_coming_soon: 0,
									} )
								}
								disabled={ isRequestingSettings }
								onClick={ eventTracker( 'Clicked Site Visibility Radio Button' ) }
								label={ translate( 'Public' ) }
							/>
						</FormLabel>
						{ isWpcomStagingSite && (
							<FormSettingExplanation>
								{ translate(
									'Your site is visible to everyone, but search engines are discouraged from indexing staging sites.'
								) }
							</FormSettingExplanation>
						) }
					</>
				) }
```

Reply:
[Question] Isn't `siteIsAtomic &&` enough instead of `! isNonAtomicJetpackSite` which is actually the inverse of `siteIsJetpack && ! siteIsAtomic`, aka `siteIsAtomic && !siteIsJetpack`?
Can a site be atomic but not be a Jetpack site?

Reply:
I would avoid changing this condition in this PR, because it's used to render the other radio buttons too.

> Can a site be atomic but not be a Jetpack site?
Not from I'm aware of. Technically a user could uninstall Jetpack, but probably that page would not be "accessible" anymore.